web/src/ResourceGroupsContext.tsx

```diff
@@ -0,0 +1,69 @@
+import { createContext, PropsWithChildren, useContext } from "react"
+import { AnalyticsAction, AnalyticsType, incr } from "./analytics"
+import { usePersistentState } from "./LocalStorage"
+
+const GROUPS_LOCAL_STORAGE_NAME = "resource-groups"
+const DEFAULT_GROUP_STATE = true
+
+export type GroupExpandedState = { [key: string]: boolean }
+
+type ResourceGroupsContext = {
```

Comment:
It seems extremely surprising to me that a method named getGroup/setGroup does collapse/expand? I feel like this file needs a naming/comment revamp (as well as tests)

Reply:
ah, yeah! 🤦🏻 my thinking was that we might want to track other properties (besides an `expanded` state) related to groups in the future, but i didn't want to over-engineer by creating a more complex state than we need right now. i can make these methods/state clearer and add comments.

Reply:
i will address this in a separate pr with the persistent state code!